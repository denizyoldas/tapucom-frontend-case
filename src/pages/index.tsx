import type { NextPage } from 'next'

const Home = () => {
  return (
    <div>
      <h1 className="text-lg font-bold">Account</h1>
    </div>
  )
}

export default Home
