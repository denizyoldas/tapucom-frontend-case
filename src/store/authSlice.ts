import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { HYDRATE } from 'next-redux-wrapper'
import { AppState } from '.'

export interface AuthState {
  authState: boolean
}

const initialState: AuthState = {
  authState: false
}

export const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    setAuthState: (state, action: PayloadAction<boolean>) => {
      state.authState = action.payload
    }
  },
  extraReducers: {
    [HYDRATE]: (state, action) => {
      return {
        ...state,
        ...action.payload.auth
      }
    }
  }
})

export const { setAuthState } = authSlice.actions

export const selectAuthState = (state: AppState) => state.auth.authState

export default authSlice.reducer
