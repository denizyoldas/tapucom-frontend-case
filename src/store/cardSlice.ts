import { HYDRATE } from 'next-redux-wrapper'
import { PayloadAction, createSlice } from '@reduxjs/toolkit'
import { IProduct } from '@/types/common.model'
import { AppState } from '.'
import { nanoid } from 'nanoid'

const PRODUCTS = [
  {
    id: nanoid(),
    name: 'Villa Bosphorus',
    description: '2 Bedrooms, 1 Bathroom',
    image: '/images/product.png',
    rating: 5.0,
    price: 1_000_000,
    distance: 3,
    isAdded: false
  },
  {
    id: nanoid(),
    name: 'Villa Bosphorus',
    description: '2 Bedrooms, 1 Bathroom',
    image: '/images/product.png',
    rating: 5.0,
    price: 1_000_000,
    distance: 3,
    isAdded: true
  },
  {
    id: nanoid(),
    name: 'Villa Bosphorus',
    description: '2 Bedrooms, 1 Bathroom',
    image: '/images/product.png',
    rating: 5.0,
    price: 1_000_000,
    distance: 3,
    isAdded: false
  },
  {
    id: nanoid(),
    name: 'Villa Bosphorus',
    description: '2 Bedrooms, 1 Bathroom',
    image: '/images/product.png',
    rating: 5.0,
    price: 1_000_000,
    distance: 3,
    isAdded: false
  },
  {
    id: nanoid(),
    name: 'Villa Bosphorus',
    description: '2 Bedrooms, 1 Bathroom',
    image: '/images/product.png',
    rating: 5.0,
    price: 1_000_000,
    distance: 3,
    isAdded: false
  }
]

export interface CardSlice {
  items: IProduct[]
  quantity: number
  total: number
}

export const initialState: CardSlice = {
  items: PRODUCTS,
  quantity: 0,
  total: 0
}

export const cardSlice = createSlice({
  name: 'card',
  initialState,
  reducers: {
    addCardItem: (state, action: PayloadAction<string>) => {
      const item = state.items.find(item => item.id === action.payload)
      if (item) {
        state.quantity++
        state.total += item.price
        item.isAdded = true
      }
    },
    removeCardItem: (state, action: PayloadAction<string>) => {
      const item = state.items.find(item => item.id === action.payload)
      if (item) {
        state.quantity--
        state.total -= item.price
        item.isAdded = false
      }
    }
  },
  extraReducers: {
    [HYDRATE]: (state, action) => {
      return {
        ...state,
        ...action.payload.auth
      }
    }
  }
})

export const { addCardItem, removeCardItem } = cardSlice.actions

export const selectCardItems = (state: AppState) => state.card.items

export default cardSlice.reducer
