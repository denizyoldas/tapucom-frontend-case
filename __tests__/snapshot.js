import { render } from '@testing-library/react'
import Account from '../src/pages/account'
import { wrapper } from '../src/store'

it('renders Account unchanged', () => {
  const { container } = render(wrapper.withRedux(<Account />))
  expect(container).toMatchSnapshot()
})
