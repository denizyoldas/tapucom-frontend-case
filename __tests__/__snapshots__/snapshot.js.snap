// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Account unchanged 1`] = `<div />`;
